db/multi_test.go

```diff
@@ -7,23 +7,34 @@ import (
 )
 
 func TestNewMulti(t *testing.T) {
-	_, err := NewMultiInserter("", "colA", "")
+	_, err := NewMultiInserter("", []string{"colA"}, "")
 	test.AssertError(t, err, "Empty table name should fail")
 
-	_, err = NewMultiInserter("myTable", "", "")
-	test.AssertError(t, err, "Empty fields string should fail")
+	_, err = NewMultiInserter("myTable", nil, "")
+	test.AssertError(t, err, "Empty fields list should fail")
 
-	mi, err := NewMultiInserter("myTable", "colA", "")
+	mi, err := NewMultiInserter("myTable", []string{"colA"}, "")
 	test.AssertNotError(t, err, "Single-column construction should not fail")
-	test.AssertEquals(t, mi.numFields, 1)
+	test.AssertEquals(t, len(mi.fields), 1)
 
-	mi, err = NewMultiInserter("myTable", "colA,colB,  colC", "")
+	mi, err = NewMultiInserter("myTable", []string{"colA", "colB", "colC"}, "")
 	test.AssertNotError(t, err, "Multi-column construction should not fail")
-	test.AssertEquals(t, mi.numFields, 3)
+	test.AssertEquals(t, len(mi.fields), 3)
+
+	_, err = NewMultiInserter("", []string{"colA"}, "colB")
+	test.AssertError(t, err, "expected error for empty table name")
+	_, err = NewMultiInserter("foo\"bar", []string{"colA"}, "colB")
+	test.AssertError(t, err, "expected error for invalid table name")
+
+	_, err = NewMultiInserter("myTable", []string{"colA", "foo\"bar"}, "colB")
+	test.AssertError(t, err, "expected error for invalid column name")
+
+	_, err = NewMultiInserter("myTable", []string{"colA"}, "foo\"bar")
+	test.AssertError(t, err, "expected error for invalid returning column name")
 }
```

Comment:
> Identifiers are stored as Unicode (UTF-8)

This is fine.

> Identifiers may or may not be case-sensitive. See [Indentifier Case-sensitivity](https://mariadb.com/kb/en/identifier-case-sensitivity/).

Nothing for us to check for here.

> Database, table and column names can't end with space characters

We can check for this.

> Identifier names may begin with a numeral, but can't only contain numerals unless quoted.

We can check for this.

> An identifier starting with a numeral, followed by an 'e', may be parsed as a floating point number, and needs to be quoted.

We can check for this.

> Identifiers are not permitted to contain the ASCII NUL character (U+0000) and supplementary characters (U+10000 and higher).

This is already checked for indirectly.

> Names such as 5e6, 9e are not prohibited, but it's strongly recommended not to use them, as they could lead to ambiguity in certain contexts, being treated as a number or expression.

Since we're checking for unquoted-allowed names, we would reject these based on the check above ("starting with a numeral, followed by an 'e'").

> User variables cannot be used as part of an identifier, or as an identifier in an SQL statement.

Huh, I don't know what this one means.